kythe/go/platform/kzip/info/info.go

```diff
@@ -47,29 +52,34 @@ func KzipInfo(f kzip.File, scanOpts ...kzip.ScanOption) (*apb.KzipInfo, error) {
 	err := kzip.Scan(f, func(rd *kzip.Reader, u *kzip.Unit) error {
 		kzipInfo.TotalUnits++
 
+		srcs := stringset.New(u.Proto.SourceFile...)
 		// The corpus may be specified in the unit VName or in the source file
 		// VNames. Record all values of corpus seen and afterwards check that a
 		// single value is specified.
 		var corpora stringset.Set
-		if u.Proto.GetVName().GetCorpus() != "" {
-			corpora.Add(u.Proto.GetVName().GetCorpus())
+		cuCorpus := u.Proto.GetVName().GetCorpus()
+		if cuCorpus != "" {
+			corpora.Add(cuCorpus)
+		} else {
+			log.Printf("Warning: Corpus not set for compilation unit %v", u.Proto.GetVName())
+			cuCorpus = unknownCorpus
 		}
-		srcs := stringset.New(u.Proto.SourceFile...)
+		corpusInfo(cuCorpus).SourceFiles[u.Proto.GetVName().GetLanguage()] += int32(srcs.Len())
+		kzipInfo.TotalSourceFiles += int32(srcs.Len())
+		corpusInfo(cuCorpus).Units[u.Proto.GetVName().GetLanguage()]++
 
 		for _, ri := range u.Proto.RequiredInput {
-			kzipInfo.TotalFiles++
+			kzipInfo.TotalRequiredInputFiles++
 			// Determine language from the unit VName (note that file VNames are
 			// forbidden from specifying a language).
-			corpusInfo(ri.GetVName().GetCorpus()).Files[u.Proto.GetVName().GetLanguage()]++
+			corpusInfo(ri.GetVName().GetCorpus()).RequiredInputFiles[u.Proto.GetVName().GetLanguage()]++
```

Comment:
This is weird.  We're counting source files toward the compilation unit's corpus, but we're counting required inputs toward their specific VName corpus.  Nowhere do we record the number of required inputs for the compilation unit's corpus and the difference isn't documented anywhere.

Reply:
It is odd. I tried to make it less odd by having a field that counts the CUs and the sources and required inputs in them and a different field to count the required inputs by their own corpus in their VName.

Reply:
discussed offline and cleaned up to only count corpus and language from the CU.